packages/oidc-http-server-pages/src/create-templates.ts

```diff
@@ -0,0 +1,111 @@
+// Helper script to prepare templates of the
+// OIDC plugin pages. This script is run during package compilation.
+//
+// By generating static pages, we can avoid having all dependencies
+// of this package pulling in react, react-dom, leafygreen, compass-components, etc.
+//
+// Furthermore, the templates are consumable by other languages
+// https://jira.mongodb.org/browse/COMPASS-7646
+//
+
+import { renderToStaticMarkup } from 'react-dom/server';
+import { renderStylesToString } from '@leafygreen-ui/emotion';
+import {
+  OIDCAcceptedPage,
+  OIDCErrorPage,
+  OIDCNotFoundPage,
+} from './pages-source';
+import React from 'react';
+import type { PageTemplates, ITemplate } from './types';
+import { HttpServerPage } from './types';
+
+/** Iterate all sub-arrays of an array */
+function* allSubsets<T>(array: T[]): Iterable<T[]> {
+  if (array.length === 0) {
+    yield [];
+    return;
+  }
+  const first = array[0];
+  for (const slicedSubset of allSubsets(array.slice(1))) {
+    yield [first, ...slicedSubset];
+    yield slicedSubset;
+  }
+}
+
+function placeholder(prop: string): string {
+  return `{{prop:${prop}}}`;
+}
+
+type Component = React.FunctionComponent<Partial<Record<string, string>>> & {
+  name: string;
+};
+
+function getPageTemplates({
+  Component,
+  parameters,
+}: {
+  Component: Component;
+  parameters: string[];
+}): ITemplate[] {
+  const templates: ITemplate[] = [];
+  for (const paramsSubset of allSubsets(parameters)) {
+    const propsObject = paramsSubset.reduce((obj, prop) => {
+      obj[prop] = placeholder(prop);
+      return obj;
+    }, {} as Record<string, string>);
```

Comment:
This is the same as doing

```suggestion
    const propsObject = Object.fromEntries(
      paramsSubset.map((prop) => [prop, placeholder(prop)])
    );
```

right? No strong feelings, but I personally find that a bit simpler to read

Reply:
yes it's the same, sure!